# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: xpcom/build/FileLocation.cpp
# Commit: 1bdd52fa9b1c
# Full Hash: 1bdd52fa9b1c0746ad6738c4dd79d571419ec4aa
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-03-31 09:35:41
# Description:
#   Bug 1761667: Handle failed OpenArchive in FileLocation r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142483
# ==============================================================================

diff -r 0dbe666a3aa6 -r 1bdd52fa9b1c xpcom/build/FileLocation.cpp
--- a/xpcom/build/FileLocation.cpp	Wed Mar 30 18:09:28 2022 +0000
+++ b/xpcom/build/FileLocation.cpp	Wed Mar 30 18:30:17 2022 +0000
@@ -151,11 +151,14 @@
   }
   aData.mZip = mBaseZip;
   if (!aData.mZip) {
+    // this can return nullptr
     aData.mZip = nsZipArchive::OpenArchive(mBaseFile);
   }
-  aData.mItem = aData.mZip->GetItem(mPath.get());
-  if (aData.mItem) {
-    return NS_OK;
+  if (aData.mZip) {
+    aData.mItem = aData.mZip->GetItem(mPath.get());
+    if (aData.mItem) {
+      return NS_OK;
+    }
   }
   return NS_ERROR_FILE_UNRECOGNIZED_PATH;
 }
